enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,404 @@
+availability. The current SDN migration solution, that will cause a service
+interruption, is not acceptable.
+
+### Goals
+
+- Migrate the cluster network provider from OpenShift SDN to OVN Kubernetes for a 
+  existing cluster.
+  - The OVN Kubernetes will be running in the multi-zone IC mode.
+- This solution shall work in all platforms managed by the SD team.
+  - ARO (Azure Red Hat OpenShift)
+  - OSDv4 (OpenShift Dedicated) on GCP
+  - OSDv4 (OpenShift Dedicated) on AWS
+  - ROSA Classic (Red Hat OpenShift Service on AWS)
+- This is in-place migration without requiring extra nodes.
+- The impact to workload of the migration shall be similar as an OCP upgrade.
+- The solution can work in scale, e.g. in a large cluster with 100+ nodes.
+- The migration operation shall be able to be rolled back if needed.
+
+### Non-Goals
+
+- Support for migration to other network providers
+- The necessary GUI change in Openshift Cluster Manager.
+- Ensure the following features remain working during live migration:
+  - Egress IP
+  - Egress Router
+  - Egress Firewall
+  - Multicast
+- Migration from SDN Multi-tenant mode
+- Migrate to OVN Kubernetes single-zone IC mode.
+- Support Hypershift clusters
+
+## Pre-requisites
+
+- OVN-IC multi-zone mode in OCP is dev-complete.
+
+## Proposal
+
+The key problem of do a live SDN migration is that, we need to the connectivity 
+of the cluster network during the migration, when pods are attached to different
+networks. We propose to utilize the OVN Kubernetes hybrid overlay feature to
+connect the networks owned by OpenShift SDN and OVN Kubernetes.
+
+- We will run different plugins on different nodes, but both plugins will know
+  how to reach pods owned by the other plugin, so all pods/services/etc remain
+  connected.
+- During migration, CNO will take original-plugin nodes one by one and convert
+  them to destination-plugin nodes, rebooting them in the process.
+- The cluster network CIDR will remain unchanged, so does the node host subnet
+  of each node.
+- NetworkPolicy will work correctly throughout the migration.
+
+### Limitations
+
+- The following features which are only supported by OpenShift SDN but not by
+  OVN Kubernetes, will stop working when the migration is started.
+  - Multitenant Isolation
+- The Egress Router feature is supported by both OpenShift SDN and OVN
+  Kubernetes but with different design and implementation. Therefore even with
+  the same name, it have different API and configure logic between OpenShift
+  SDN and OVN Kubernetes. Also, the supported platforms and modes are different
+  between the two network providers. So users need to evaluate before conducting
+  the live migration, and have to migrate the configuration manually after the
+  migration is complete.
+- The following features are supported by both OpenShift SDN and OVN Kubernetes.
+  And we have support automated converting for the configuration. However, users
```

Comment:
How "automated" is automated? When does this happen? (The "Workflow Description" below doesn't mention it.)

Reply:
This has already been implemented https://issues.redhat.com/browse/NP-20.